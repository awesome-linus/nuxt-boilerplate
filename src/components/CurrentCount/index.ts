import Vue from 'vue';

export default Vue.extend({
  name: 'CurrentCount',
  props: {
    count: {
      type: Number,
      default: 0,
      required: true
    }
  }
});
