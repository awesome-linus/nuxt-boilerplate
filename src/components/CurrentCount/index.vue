<template>
  <div :class="$style.currentCountArea" data-test="current-count">
    Count: {{ count }}
  </div>
</template>

<script lang="ts" src="./index.ts" />

<style lang="scss" module src="./index.module.scss" />
