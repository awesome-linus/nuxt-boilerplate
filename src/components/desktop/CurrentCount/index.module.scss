.currentCountArea {
  font-size: 40px;
}
