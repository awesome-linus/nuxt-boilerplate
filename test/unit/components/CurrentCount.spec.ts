import { shallowMount } from '@vue/test-utils';
import CurrentCount from '~/components/CurrentCount/index.vue';

describe('components/CurrentCount', () => {
  it('should be expected 1', () => {
    const countTestData = { count: 1 };
    const wrapper = shallowMount(CurrentCount, {
      propsData: { count: countTestData.count }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should be expected 3', () => {
    const countTestData = { count: 3 };
    const wrapper = shallowMount(CurrentCount, {
      propsData: { count: countTestData.count }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
