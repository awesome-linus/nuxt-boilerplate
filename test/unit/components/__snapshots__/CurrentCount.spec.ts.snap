// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/CurrentCount should be expected 1 1`] = `
"<div data-test=\\"current-count\\" class=\\"currentCountArea\\">
  Count: 1
</div>"
`;

exports[`components/CurrentCount should be expected 3 1`] = `
"<div data-test=\\"current-count\\" class=\\"currentCountArea\\">
  Count: 3
</div>"
`;
